### ISSUE TITLE
Add declarative Timer support

### ISSUE THREAD

**Issue by lordofthejars** [quarkusio/quarkus#10790]
**Description**

Currently Quarkus supports `@Schedule` annotations which let you program in a declarative way a periodic task. But sometimes what you want is to execute some method after some delayed time and just one time.

**Implementation ideas**

Something like:

[source, java]
----
@TimeTask(value=10, unit = TimeUnit.SECONDS)
public void execute() {
  sysout("I am executed after 10 seconds");
}

t.execute()
----

Then you you call `t.execute()` the method just returns immediately as it was executed but the real execution is delayed for 10 seconds.
Basically this would be an implementation of `Timer`/`TimeTask` classes in Quarkus.

One of the important things here is that the method must return `void` as the method execution happens after 10 seconds.

**Comment by mkouba** [comment]
Hm, so this is basically "asynchronous execution with a delay". I.e. something similar to `@org.eclipse.microprofile.faulttolerance.Asynchronous`. @Ladicek @michalszynkiewicz would it make sense to add such a feature to `smallrye-fault-tolerance`?

**Comment by Ladicek** [comment]
I think so, yes.

I need to do an internal refactoring to fix some annoying issues (https://github.com/smallrye/smallrye-fault-tolerance/issues/209, https://github.com/smallrye/smallrye-fault-tolerance/issues/210), and this refactoring would probably enable this feature pretty easily.

**Comment by mkouba** [comment]
Hm, I think you could introduce a new annotation like `@Delayed` so that users can still use the spec version of `@Asynchronous` annotation.

**Comment by Ladicek** [comment]
I intend to create my own annotation, tentatively called `@ThreadOffload`, because `@Asynchronous` is too ambiguous. In any case, delaying would be a separate annotation for sure, and `@Delayed` sounds good.

**Comment by mkouba** [comment]
Hm, I don't think `@ThreadOffload` is very intuitive ;-) ... but I agree that `@Asynchronous` is not good either. Maybe something like `@ExecuteOnWokerThread`... ok, it's not very intuitive either...

**Comment by michalszynkiewicz** [comment]
technically it makes sense but delaying doesn't have much in common with fault tolerance ;)

**Comment by Ladicek** [comment]
There are some [more or less] valid arguments against putting `@Asynchronous` into Fault Tolerance in the first place, but the fact of life is that all fault tolerance libraries have means to execute code on an extra thread. In which case, I'd argue it's quite natural to support 1. some rudimentary scheduling, and 2. "fire and forget" scenarios (return type of `void`).

**Comment by mkouba** [comment]
What are the "valid" arguments? I see none ;-)

**Comment by michalszynkiewicz** [comment]
Trolling aside, :+1:  on adding it to SR FT. 
Would be great to have a separate spec for it if it were to move to MP at some point.

**Comment by janmaterne** [comment]
@Delay sounds best jo me.
But you could also support returning future objects.

**Comment by gavinking** [comment]
Why is this a good thing to use annotations for? Is it really the case that "10 seconds" is always a hardcoded value and never something that is calculated from information defined elsewhere?

This, naively at least, looks to me like another example of an API designed before Java and higher-order functions.

**Comment by gavinking** [comment]
> This, naively at least, looks to me like another example of an API designed before Java had higher-order functions.

What I mean is, it's like the `Cacheable` annotation for memorization: it's perhaps defensible as a convenience sugar for something defined primitively as an API, but it shouldn't itself be the source of the primitive definition of the behavior.
